strawberry/field.py

```diff
@@ -152,32 +209,28 @@ def base_resolver(self, resolver: StrawberryResolver) -> None:
         #       removed.
         _ = resolver.arguments
 
-    @property  # type: ignore
-    def type(self) -> Union[StrawberryType, type]:  # type: ignore
-        # We are catching NameError because dataclasses tries to fetch the type
-        # of the field from the class before the class is fully defined.
-        # This triggers a NameError error when using forward references because
-        # our `type` property tries to find the field type from the global namespace
-        # but it is not yet defined.
-        try:
-            if self.base_resolver is not None:
-                # Handle unannotated functions (such as lambdas)
-                if self.base_resolver.type is not None:
-                    return self.base_resolver.type
-
-            assert self.type_annotation is not None
-
-            if not isinstance(self.type_annotation, StrawberryAnnotation):
-                # TODO: This is because of dataclasses
-                return self.type_annotation
-
-            return self.type_annotation.resolve()
-        except NameError:
-            return None  # type: ignore
-
-    @type.setter
-    def type(self, type_: Any) -> None:
-        self.type_annotation = type_
+    def get_type(self) -> Union[StrawberryType, object]:
+        if self.base_resolver is not None:
+            # Handle unannotated functions (such as lambdas)
+            if self.base_resolver.type is not None:
+                return self.base_resolver.type
+
+        assert self.type is not None
```

Comment:
At minimum, this should probably throw a more descriptive error

Reply:
Yep agreed. I copied this from the previous function